lib/backend/k8s/resources/networkpolicy.go

```diff
@@ -289,9 +288,23 @@ func (c *networkPolicyClient) EnsureInitialized() error {
 }
 
 func (c *networkPolicyClient) Watch(ctx context.Context, list model.ListInterface, revision string) (api.WatchInterface, error) {
-	resl := list.(model.ResourceListOptions)
-	if len(resl.Name) != 0 {
-		return nil, fmt.Errorf("cannot watch specific resource instance: %s", list.(model.ResourceListOptions).Name)
+	// Build watch options to pass to k8s.
+	opts := metav1.ListOptions{Watch: true}
+	rlo, ok := list.(model.ResourceListOptions)
+	if !ok {
+		return nil, fmt.Errorf("ListInterface is not a ResourceListOptions: %s", list)
+	}
+	if len(rlo.Name) != 0 {
```

Comment:
Hm, I think we only need to parse the policy name if this is a `kns.` prefixed policy, since those are backed by k8s.

There are two watches in this func, so need to handle both cases and only start one of the two watches if a name is passed (like with profiles)